src/Microsoft.PowerShell.Commands.Utility/commands/utility/join-string.cs

```diff
@@ -106,40 +106,46 @@ protected override void BeginProcessing()
         /// <inheritdoc />
         protected override void ProcessRecord()
         {
-            if (InputObject != null && InputObject != AutomationNull.Value)
+            if (InputObject != null)
```

Comment:
I think it should be:
```suggestion
            if (inputObject != null && inputObject != AutomationNull.Value)
```

Reply:
At the line 109 there is no **inputObject** variable, since it is defined on the line 111.

Reply:
Sorry...
```suggestion
            if (InputObject != null && InputObject != AutomationNull.Value)
```

Reply:
I am thinking about:
```powershell
[System.Management.Automation.Internal.AutomationNull]::Value | Join-String
Join-String -InputObject $null
```

Reply:
Isn't the first a complete no-op in any case? I don't think cmdlets receive a pipeline item that is `AutomationNull.Value`

Were you to pass it directly in as an -InputObject, though, maybe that'd get it passed in...

Reply:
null is not passed through pipeline, [AutomationNull]::Value is passed through pipeline.

Reply:
It appears both can be passed through the pipeline.
![image](https://user-images.githubusercontent.com/32407840/50906040-3a249480-13f2-11e9-9f24-83893b2d1560.png)

Reply:
@vexx32 
if using
```powershell
using namespace System.Management.Automation.Internal
Join-String -InputObject ([AutomationNull]::Value)
```
it is caught at line 109.

Reply:
Good enough for me! 😄

Reply:
This is resolved?